mlflow/promptflow/__init__.py

```diff
@@ -0,0 +1,393 @@
+        **flow_env
+    )
+
+    # append loader_module, data and env data to mlflow_model
+    pyfunc.add_to_model(
+        mlflow_model,
+        loader_module="mlflow.promptflow",
+        conda_env=_CONDA_ENV_FILE_NAME,
+        python_env=_PYTHON_ENV_FILE_NAME,
+        code=code_dir_subpath,
+    )
+
+    # save mlflow_model to path/MLmodel
+    mlflow_model.save(os.path.join(path, MLMODEL_FILE_NAME))
+
+    if conda_env is None:
+        if pip_requirements is None:
+            default_reqs = get_default_pip_requirements()
+            # To ensure `_load_pyfunc` can successfully load the model during the dependency
+            # inference, `mlflow_model.save` must be called beforehand to save an MLmodel file.
+            inferred_reqs = mlflow.models.infer_pip_requirements(
+                path, FLAVOR_NAME, fallback=default_reqs
+            )
+            default_reqs = sorted(set(inferred_reqs).union(default_reqs))
+        else:
+            default_reqs = None
+        conda_env, pip_requirements, pip_constraints = _process_pip_requirements(
+            default_reqs,
+            pip_requirements,
+            extra_pip_requirements,
+        )
+    else:
+        conda_env, pip_requirements, pip_constraints = _process_conda_env(conda_env)
+
+    with open(os.path.join(path, _CONDA_ENV_FILE_NAME), "w") as f:
+        yaml.safe_dump(conda_env, stream=f, default_flow_style=False)
+
+    # Save `constraints.txt` if necessary
+    if pip_constraints:
+        write_to(os.path.join(path, _CONSTRAINTS_FILE_NAME), "\n".join(pip_constraints))
+
+    # Save `requirements.txt`
+    write_to(os.path.join(path, _REQUIREMENTS_FILE_NAME), "\n".join(pip_requirements))
+
+    _PythonEnv.current().to_yaml(os.path.join(path, _PYTHON_ENV_FILE_NAME))
+
+
+def _resolve_env_from_flow(flow_dag_path):
+    with open(flow_dag_path, "r") as f:
+        flow_dict = yaml.safe_load(f)
+    environment = flow_dict.get("environment", {})
+    return environment
+
+
+class _PromptflowModelWrapper:
+    def __init__(self, model):
+        self.model = model
+        self.model_invoker = None
+
+    def predict(  # pylint: disable=unused-argument
+        self,
+        data: Union[pd.DataFrame, List[Union[str, Dict[str, Any]]]],
+        params: Optional[Dict[str, Any]] = None,  # pylint: disable=unused-argument
+    ):
+        """
+        :param data: Model input data.
+        :param params: Additional parameters to pass to the model for inference.
+
+                       .. Note:: Experimental: This parameter may change or be removed in a future
+                                               release without warning.
+
+        :return: Model predictions.
+        """
+        from promptflow._sdk._serving.flow_invoker import FlowInvoker
+        if self.model_invoker is None:
+            # TODO: Support more choice in model config?
+            self.model_invoker = FlowInvoker(self.model, connection_provider="local")
```

Comment:
@santiagxf any chance we can put this in inference config?

Reply:
ideally we can move this stuff into params + inference config

Reply:
(though, TBH gateway vs connection provider here is probably going to get interesting)

Reply:
we should allow customizing this connection provider here

Reply:
Added model_config to allow user set connection provider.